neonvm/controllers/virtualmachinemigration_controller.go

```diff
@@ -147,7 +155,7 @@ func (r *VirtualMachineMigrationReconciler) Reconcile(ctx context.Context, req c
 
 	// Fetch the corresponding VirtualMachine instance
 	vm := new(vmv1.VirtualMachine)
-	err := r.Get(ctx, types.NamespacedName{Name: migration.Spec.VmName, Namespace: migration.Namespace}, vm)
+	err = r.Get(ctx, types.NamespacedName{Name: migration.Spec.VmName, Namespace: migration.Namespace}, vm)
```

Comment:
This case is I think the reason why having named return parameters is less than ideal. `err` is here is not necessarily returned, so the semantics and effects of this assignment can be misleading.